messaging/src/main/protos/Messages.proto

```diff
@@ -0,0 +1,185 @@
+/**
+ * tassis uses an asynchronous messaging pattern for interacting with the API.
+ *
+ * Clients typically connect to tassis via WebSockets to exchange messages.
+ *
+ * Clients will typically open two separate connections, authenticating on one and leaving
+ * the other unauthenticated.
+ *
+ * The unauthenticated connection is used for retrieving other identities' preKeys and
+ * sending messages to them, so as not to reveal the identityKey of senders.
+ *
+ * The authenticated connection is used for all other operations, including performing key
+ * management and receiving messages from other identities.
+ *
+ * Authentication is performed using a challenge-response pattern in which the server sends
+ * an authentication challenge to the client and the client responds with a signed authentication
+ * response identifying its identityKey and deviceId. On anonymous connections, clients simply ignore the
+ * authentication challenge.
+ *
+ * Messages sent from clients to servers follow a request/response pattern. The server will always
+ * respond to these with either an Ack or a typed response. In the event of an error, it will respond
+ * with an Error message. This includes the following messages:
+ *
+ *  - Register        -> Ack
+ *  - Unregister      -> Ack
+ *  - RequestPreKeys  -> PreKeys
+ *  - OutboundMessage -> Ack
+ *
+ * Some messages sent from the server to the client require an Ack in response:
+ *
+ *  - inboundMessage  -> Ack
+ *
+ * Some messages don't require any response:
+ *
+ *  - PreKeysLow
+ *
+ * All messages sent within a given connection are identified by a unique sequence number (separate sequences
+ * for each direction). When a response message is sent in either direction, its sequence number is set to the
+ * message that triggered the response so that the client or server can correlate responses with requests.
+ */
+syntax = "proto3";
+
+package tassis;
+
+option go_package = "github.com/getlantern/tassis/model";
```

Comment:
What's wrong with `model`?

Reply:
It's generic; it doesn't convey any meaning about the contents.  This may not be _the_ model for importers outside of the tassis world.  Other importers may have their own models for storing messages and other constructs defined in `model`. 
 As an analogy, it's like having a protocol library with subpackages `client` and `server`.  Importers can always alias it as `tassismodel` or whatever they like, but then everyone is referring to your package differently.

If you're particularly tied to `model`, I won't fight you on it, but I do think it makes for awkward usage.

Reply:
Ah, now I understand. Let me think about that.